src/main/java/net/neoforged/neoforge/event/level/BlockEvent.java

```diff
@@ -464,4 +465,49 @@ public BlockState getFinalState() {
             return state;
         }
     }
+
+    /**
+     * Fired when checking if a tool can break a block.
+     * Allows for force-allowing or force-denying a tool to break a block,
+     * and for additional checks depending on additional context for the block (e.g. block entity context).
+     * <p>
+     * This event has a {@link Result}.
+     * If the result is {@link Result#ALLOW}, the tool will be considered correct for breaking the block.
+     * If the result is {@link Result#DENY}, the tool will not be considered correct for breaking the block.
+     * If the result is {@link Result#DEFAULT}, further vanilla checks will run to determine whether the tool is considered correct for breaking the block;
+     * this is vanilla behavior.
+     */
+    public static class ToolCheck extends BlockEvent {
```

Comment:
if you have a result you need to annotate with `@HasResult`

Reply:
HasResult is marked as deprecated, but will add.